### PR TITLE
Rework redis config

### PR DESCRIPTION
**Pull request by aogereau**
## Description
Completely reworked the way redis configuration is handled .
`community.general.redis` module is used to ensure proper configuration is applied . No restarts are needed .
Added additionnal testing cases for custom config and use of password

### Redis server :
redis server templates have been reconsolidated into one, and all default values and comments have been removed
Now most of the custom config is handled in a `redis_conf` dictionnary . 
exception for databases parameter as it is immutable and can't be changed via redis-cli, modifying it will require a restart

### Redis Sentinel :
Added tasks that set master back to master node if a failover has occured
Changed the way the Sentinel configuration is handled. It will still need a manual restart afterward .
Sentinel configuration is handled by three different dictionnary depending on the syntax of the configuration parameter
- redis_sentinel_conf for `<key> <value>`
- redis_sentinel_sentinel_conf for `sentinel <key> <value>`
- redis_sentinel_master_conf for `sentinel <key> <sentinel-master> <value>`

## Motivation and Context
Changing configuration would require a restart of redis .
Sentinel config rewrite could interfere, resulting in misapplied configuration
Also a lot of parameters weren't handled properly, which required additionnal commits
The way the sentinel configuration was handled, it couldn't be modified after intall

## How Has This Been Tested?
Molecule

## Types of Changes

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: vars/main.yml [hunk 1]
```diff
@@ -10,3 +10,5 @@ _redis_master_host: >-
 _redis_master_ip: "{{ hostvars[inventory_hostname]['ansible_default_ipv4']['address'] if _redis_master_host == '' else (hostvars[_redis_master_host]['ansible_all_ipv4_addresses'] | first) }}"
 
 _config_version: "{% if redis_server_version == 'latest' or redis_server_version is version('6:7.2', '>=' ) %}7.2{% elif redis_server_version is version('6:6.2', '<' ) %}6.0{% elif redis_server_version is version('6:7.0', '<' ) %}6.2{% elif redis_server_version is version('6:7.2', '<' ) %}7.0{% endif %}"
+
+_redis_conf_final: "{{ redis_conf_default |combine(redis_conf) | dict2items | rejectattr('value', 'equalto', '') | list | items2dict }}"
```
Comment (ElieDeloumeau) [inline vars/main.yml]:
```suggestion
_redis_conf_final: "{{ redis_conf_default | combine(redis_conf) | dict2items | rejectattr('value', 'equalto', '') | list | items2dict }}"
```

File: tasks/server.yml [hunk 2]
```diff
@@ -47,3 +46,36 @@
     name: redis-server@{{ redis_server_port }}.service
     state: started
   register: _redis_server_systemctl_start
+
+- name: ensure redis conf is set with password
+  community.general.redis:
+    command: config
+    name: "{{ item.key }}"
+    value: "{{ item.value }}"
+    login_host: 127.0.0.1
```
Comment (ElieDeloumeau) [inline tasks/server.yml]: `login_host` should be overridable or take care of `_redis_conf_final.bind`

File: molecule/password/tests/test_default.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+
+def test_file_config(host):
+    file = host.file("/etc/redis/redis_6379.conf")
+    assert file.exists
+    assert file.user == "redis"
+
+
+def test_redis_listening(host):
+    assert host.socket("tcp://0.0.0.0:6379").is_listening
```
Comment (ElieDeloumeau) [inline molecule/password/tests/test_default.py:19]: Please add a test to check if the password is configured

**Review by ElieDeloumeau — CHANGES_REQUESTED** [review]
Thank you for this usefull PR!

There are a few small changes to do before merging

**Comment by aogereau** [comment]
added rework for Sentinel

**Review by ElieDeloumeau — APPROVED** [review]

**Pull request merged**